### ISSUE TITLE
Error in AssemblyEvidenceSource  

### ISSUE THREAD

**Issue by cshneider** [PapenfussLab/gridss#81]
I am running with the somatic.sh script with the latest 1.3.4 GRIDSS release, having downloaded the binary version. As a test case, I ran GRIDSS on both chromosome 21 and chromosome 1 of the human genome.

My modified version of the somatic.sh script ran on a sge cluster for which I reserved 1.5 hrs (chr. 21) and 4 hrs (chr. 1) and 50G of memory and 12 threads.

The following modules were loaded:

module load Java/1.8.0_60
module load aligners/bwa/0.7.13
module load R/3.3.0
module load bed/bedtools/2.25.0

For the java command section of the script:

java -ea -Xmx31g \
   -Dgridss.defensiveGC=true \
   -Dsamjdk.create_index=true \
   -Dsamjdk.use_async_io_read_samtools=true \
   -Dsamjdk.use_async_io_write_samtools=true \
   -Dsamjdk.use_async_io_write_tribble=true \
   -Dsamjdk.compression_level=1 \    
   -cp $GRIDSS_JAR gridss.CallVariants \
   TMP_DIR=/hpc/tmp \
   WORKER_THREADS=12 \
   WORKING_DIR="$PWD" \
   REFERENCE_SEQUENCE="$REFERENCE" \
   INPUT="$INPUT" \
   OUTPUT="$OUTPUT" \
   ASSEMBLY="$ASSEMBLY" \
   BLACKLIST="$BLACKLIST" \
   2>&1 | tee -a gridss.$HOSTNAME.$$.log

From the error log, 'terminateOnFirstError=true'.
The following error results: AssemblyEvidenceSource, Error assembling chunk 290  (21:10000001-20000000).
java.lang.IllegalStateException
at com.google.common.base.Preconditions.checkState(Preconditions.java:159)
at com.google.common.collect.AbstractIterator.hasNext(AbstractIterator.java:130)
at com.google.common.collect.Iterators$PeekingImpl.hasNext(Iterators.java:1171)
at au.edu.wehi.idsv.debruijn.positional.PositionalAssembler.ensureAssembler(PositionalAssembler.java:92)
at au.edu.wehi.idsv.debruijn.positional.PositionalAssembler.hasNext(PositionalAssembler.java:54)
at au.edu.wehi.idsv.AssemblyEvidenceSource.assembleChunk(AssemblyEvidenceSource.java:245)
at au.edu.wehi.idsv.AssemblyEvidenceSource.assembleChunk(AssemblyEvidenceSource.java:205)
at au.edu.wehi.idsv.AssemblyEvidenceSource.lambda$assembleBreakends$15(AssemblyEvidenceSource.java:84)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
at java.lang.Thread.run(Thread.java:745)

Right before this error, the following Warning reoccurs several time:  
SplitReadEvidence  Read ST-E00290:28:HF3WLCCXX:7:2115:15128:53926 is hard clipped. Please run gridss.ComputeSamTags to soften hard clips. This message will be displayed 0 more times.  
However, gridss.ComputeSamTags is actually run as stated in the log file.

What would be the proper steps you would advise to follow? 
Is it also possible to set 'terminateOnFirstError=false' and would this be done in the Java command section?

**Comment by d-cameron** [comment]
Are you able to supply the initial underlying error message from the log file? That error message indicates that auto-recovery was unsuccessful, but does not actually identify the root cause.

**Comment by d-cameron** [comment]
>Right before this error, the following Warning reoccurs several time:
SplitReadEvidence Read ST-E00290:28:HF3WLCCXX:7:2115:15128:53926 is hard clipped. Please run gridss.ComputeSamTags to soften hard clips. This message will be displayed 0 more times.
However, gridss.ComputeSamTags is actually run as stated in the log file.

This will be cause by split reads that your single-chromosome filter has removed one of the split alignments from. For example, a split read which is aligned to both chr1 and chr2 will contain two records in the bam file (one at each location). Without the primary alignment record, gridss.ComputeSamTags is unable to convert the hard clipped supplementary alignment to a soft clipped bases since they don't exist in the input file you supplied to GRIDSS.

TLDR: the soft-clipping warning message is expected given your filtering, but unrelated to the underlying problem causing GRIDSS to crash.

**Comment by cshneider** [comment]
Dear Daniel, 

Thanks for your reply. 

The log file displays no other ERROR messages aside from:
Error assembling chunk 290 (21:10000001-20000000)
There are no messages that contain the example messages "This should not happen", "Please raise an issue at", or  "Attempting recovery". 

However, there are multiple WARNING messages referring to missing alignments. 
Would my understanding be correct that it is not possible to test on a single chromosome because all alignments need to be contained in the input .bam file and this is only possible with the full genome? 

Thanks for your assistance, 
Carl

**Comment by d-cameron** [comment]
It is quite strange as that error message should only occur after an underlying error is thrown. Can you try with GRIDSS 1.4.0?

Thanks

**Comment by cshneider** [comment]
Dear Daniel, 

Thanks for your reply! 
Running GRIDSS 1.3.4 on the entire human genome worked! 
I will also run GRIDSS 1.4.0 now. 

Cheers,
Carl
